### ISSUE TITLE
Func contract's return value isn't enforced with blocks

### ISSUE THREAD

**Issue by indigoviolet** [egonSchiele/contracts.ruby#242]
```
Contract Func[Num=>Num] => nil
def foo(&blk)
  result = blk.call(2)
  nil
end

Contract Func[Num=>Num] => nil
def bar(blk)
  result = blk.call(2)
  nil
end

foo { |x| x.to_s }  # no error
bar lambda { |x| x.to_s } # error
```

**Comment by egonSchiele** [comment]
I can reproduce

**Comment by egonSchiele** [comment]
This is fixed in v0.15.
